benchmarks/prescribe_dike_injection/prescribed_dike_injection_heat.cc

```diff
@@ -0,0 +1,137 @@
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    LatentHeatDikeInjection<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Heating model");
+      {
+        prm.enter_subsection("Latent heat dike injection");
+        {
+          prm.declare_entry ("Latent heat of crystallization", "1.1e9",
+                             Patterns::Double(0),
+                             "The latent heat of crystallization that is released when material "
+                             "is injected into the model. "
+                             "Units: J/m$^3$.");
+          prm.declare_entry ("Temperature of the injected material", "1273",
+                             Patterns::Double(0),
+                             "The temperature of the material injected into the model. "
+                             "Units: K.");
+          prm.declare_entry("Dike material injection fraction", "0.0",
+                            Patterns::Double(0),
+                            "Amount of new injected material from the dike. Units: none.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    LatentHeatDikeInjection<dim>::
+    create_additional_material_model_outputs(MaterialModel::MaterialModelOutputs<dim> &material_model_outputs) const
+    {
+      this->get_material_model().create_additional_named_outputs(material_model_outputs);
+    }
+
+    template <int dim>
+    void
+    LatentHeatDikeInjection<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Heating model");
+      {
+        prm.enter_subsection("Latent heat dike injection");
+        {
+          latent_heat_of_crystallization = prm.get_double ("Latent heat of crystallization");
+          temperature_of_injected_material = prm.get_double ("Temperature of the injected material");
+          dike_material_injection_fraction = prm.get_double ("Dike material injection fraction");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace HeatingModel
+  {
+    ASPECT_REGISTER_HEATING_MODEL(LatentHeatDikeInjection,
+                                  "latent heat dike injection",
+                                  "Latent heat releases due to the material injection (e.g., melt) into the model. "
+                                  "This heating model takes the source term added to the Stokes "
+                                  "equation and adds the corresponding source term to the energy "
+                                  "equation. This source term includes both the effect of latent "
+                                  "heat release upon crystallization and the fact that injected "
+                                  "material might have a different temperature.")
```

Comment:
If I remember correctly, in the corresponding prm, you give the equations for the heating term. They would go here nicely as well!